fft/adi_fft.c

```diff
@@ -592,8 +592,9 @@ int adi_fft_perform(struct adi_fft_processing *fft_proc,
 	if (ret)
 		return ret;
 
+	/* Convert codes without DC offset to "volts" without respect to Vref voltage */
 	sample_cnt = 0;
-	for (cnt = 0; cnt < fft_proc->fft_length * 4; cnt+=2) {
+	for (cnt = 0; cnt < fft_proc->fft_length * 2; cnt+=2) {
```

Comment:
Any reason behind this change?

Reply:
so for total samples of 4096, CFFT is performed on 2048 real samples and remaining 2048 samples are imaginary (which are set as 0). fft_length therefore is 2048. With fft_length * 4 i.e. 8192, the number of real and img samples becomes 4096 each. Looking at DSP FFT library implementation, the FFT analysis is done by looping through fft_length*2 (4096) iterations. Hence it make sense to just get 4096 samples and not 8192.
There are some issues in FFT calculations which needs to be rectified. So I am still working on this and in future there might be few more changes